assignments/closure.js

```diff
@@ -1,6 +1,12 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
 
+var yomama= 'fat'
+ function snaps (){
+   console.log(`Yo mama is so fat ,` snaps + '!')
+ }
+fat = 'when she got on the scale it said,"I need your weight not your phone number'
+ snaps()
```

Comment:
Ok so don't use `var` anymore, please :)

Closure is like this:
```
// This is a let because it may change in the future
let punchline = 'when she got on the scale it said,"I need your weight not your phone number"'

function snaps() { // Function - outer
  return function(punchSnap){ // Function - inner, with parameter
    return `Yo mama is so ${whatIsYoMama}, ${punchSnap}!`
  }  
}

// This is a const because mama is always gonna be fat
const whatIsYoMama = "fat" 

// This is the closure. When you assign a function to a variable, thereby giving access to the 
// function inside of the function to the variable.
const closureSnap = snaps()

// Here we invoke the variable as a function and it get's passed to the inner function 
// as the punchSnap parameter
// punchline is the variable we defined above
console.log(closureSnap(punchline))

```